examples/all-clusters-app/linux/main-common.cpp

```diff
@@ -205,142 +214,116 @@ void HandleSwitchEvent(intptr_t arg)
+    }
+    else if (strncmp("LongRelease", payload, strlen("LongRelease")) == 0)
+    {
+        HandleSwitchEvent(Clusters::Switch::Events::LongRelease::Id);
+    }
+    else if (strncmp("MultiPressOngoing", payload, strlen("MultiPressOngoing")) == 0)
+    {
+        HandleSwitchEvent(Clusters::Switch::Events::MultiPressOngoing::Id);
+    }
+    else if (strncmp("MultiPressComplete", payload, strlen("MultiPressComplete")) == 0)
+    {
+        HandleSwitchEvent(Clusters::Switch::Events::MultiPressComplete::Id);
+    }
+    else if (strncmp("PowerOnReboot", payload, strlen("PowerOnReboot")) == 0)
+    {
+        HandleRebootSignal(BootReasonType::kPowerOnReboot);
+    }
+    else if (strncmp("BrownOutReset", payload, strlen("BrownOutReset")) == 0)
+    {
+        HandleRebootSignal(BootReasonType::kBrownOutReset);
+    }
+    else if (strncmp("SoftwareWatchdogReset", payload, strlen("SoftwareWatchdogReset")) == 0)
+    {
+        HandleRebootSignal(BootReasonType::kSoftwareWatchdogReset);
+    }
+    else if (strncmp("HardwareWatchdogReset", payload, strlen("HardwareWatchdogReset")) == 0)
+    {
+        HandleRebootSignal(BootReasonType::kHardwareWatchdogReset);
+    }
+    else if (strncmp("SoftwareUpdateCompleted", payload, strlen("SoftwareUpdateCompleted")) == 0)
+    {
+        HandleRebootSignal(BootReasonType::kSoftwareUpdateCompleted);
+    }
+    else if (strncmp("SoftwareReset", payload, strlen("SoftwareReset")) == 0)
+    {
+        HandleRebootSignal(BootReasonType::kSoftwareReset);
+    }
+    else
+    {
+        ChipLogError(NotSpecified, "Unhandled command: Should never happens");
+    }
 }
 
-void OnGeneralFaultSignalHandler(int signum)
+void * EventCommandListenerTask(void * arg)
 {
-    ChipLogDetail(DeviceLayer, "Caught signal %d", signum);
+    int fd;
+    char readbuf[CHIP_EVENT_COMMAND_SIZE];
+    ssize_t readBytes;
 
-    uint32_t eventId;
-    switch (signum)
+    for (;;)
     {
-    case SIGUSR2:
-        eventId = Clusters::GeneralDiagnostics::Events::HardwareFaultChange::Id;
-        break;
-    case SIGHUP:
-        eventId = Clusters::GeneralDiagnostics::Events::RadioFaultChange::Id;
-        break;
-    case SIGTTIN:
-        eventId = Clusters::GeneralDiagnostics::Events::NetworkFaultChange::Id;
-        break;
-    default:
-        ChipLogError(NotSpecified, "Unhandled signal: Should never happens");
-        chipDie();
-        break;
-    }
+        fd = open(CHIP_EVENT_FIFO_PATH, O_RDONLY);
 
-    PlatformMgr().ScheduleWork(HandleGeneralFaultEvent, static_cast<intptr_t>(eventId));
-}
+        readBytes          = read(fd, readbuf, CHIP_EVENT_COMMAND_SIZE);
+        readbuf[readBytes] = '\0';
+        ChipLogProgress(Zcl, "Received payload: \"%s\" and length is %ld\n", readbuf, readBytes);
 
-void OnSwitchSignalHandler(int signum)
-{
-    ChipLogDetail(DeviceLayer, "Caught signal %d", signum);
+        // Process the received command request from event fifo
+        ProcessCommandRequest(readbuf);
```

Comment:
So what happens for future compatibility? What triggers the `read` to return? I don't see any code to check for a delimiter

Reply:
Currently, the payload only contain the event command name itself, no delimiter, the plan is to move payload to json for future extensibility. This will be done in a separate PR. https://github.com/project-chip/connectedhomeip/issues/20594